backend/supabase/group_invites/create.sql

```diff
@@ -0,0 +1,45 @@
+drop table group_invites;
+
+create table if not exists
+  group_invites (
+    id text not null primary key default random_alphanumeric (12),
+    group_id text not null,
+    foreign key (group_id) references groups (id),
+    created_time timestamptz not null default now(),
+    duration interval default '1 week',
+    is_forever boolean not null default false,
+    check (
+      (
+        duration is null
+        and is_forever = true
+      )
+      or (duration is not null)
+    ),
+    uses numeric not null default 0,
+    max_uses numeric default null,
+    is_max_uses_reached boolean generated always as (
+      case
+        when max_uses is null then false
+        else uses >= max_uses
+      end
+    ) stored,
+    expire_time timestamptz
+  );
+
+  CREATE OR REPLACE FUNCTION set_expire_time()
```

Comment:
Doesn't auto-format change to lowercase? Would be nice to check in only consistent formats.

Reply:
Seems like this could also be generated? Or is there a reason to prefer using a trigger?

Reply:
yeah I was not able to make a generated column, because it adds something to the current time so it's "non-deterministic"